src/tests/unit/tests/background/dev-tools-monitor.test.ts

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { OnDevToolStatusPayload } from 'background/actions/action-payloads';
+import { DevToolsMonitor } from 'background/dev-tools-monitor';
+import { TabContextManager } from 'background/tab-context-manager';
+import { BrowserAdapter } from 'common/browser-adapters/browser-adapter';
+import { Messages } from 'common/messages';
+import {
+    DelayCreator,
+    PromiseFactory,
+    TimeoutCreator,
+    TimeoutError,
+} from 'common/promises/promise-factory';
+import _ from 'lodash';
+import { flushSettledPromises } from 'tests/common/flush-settled-promises';
+import { IMock, It, Mock, Times } from 'typemoq';
+import { DictionaryNumberTo } from 'types/common-types';
+
+class TestDevToolsMonitor extends DevToolsMonitor {
+    public activeDevtoolTabIds: number[];
+
+    public async testPollLoop(): Promise<void> {
+        await super.monitorActiveDevtools();
+    }
+}
+
+describe(DevToolsMonitor, () => {
+    const messageTimeout = 10;
+    const pollInterval = 20;
+    const tabId = 1;
+    const anotherTabId = 10;
+
+    const messageSuccessResponse = { isActive: true };
+    const mockTimeoutResponse = { isTimeout: true };
+
+    let browserAdapterMock: IMock<BrowserAdapter>;
+    let timeoutMock: IMock<TimeoutCreator>;
+    let delayMock: IMock<DelayCreator>;
+    let promiseFactory: PromiseFactory;
+    let tabContextManagerMock: IMock<TabContextManager>;
+
+    let testSubject: TestDevToolsMonitor;
+
+    beforeEach(() => {
+        browserAdapterMock = Mock.ofType<BrowserAdapter>();
+        timeoutMock = Mock.ofType<TimeoutCreator>();
+        delayMock = Mock.ofType<DelayCreator>();
+        promiseFactory = {
+            timeout: timeoutMock.object,
+            delay: delayMock.object,
+        } as PromiseFactory;
+        tabContextManagerMock = Mock.ofType<TabContextManager>();
+
+        testSubject = new TestDevToolsMonitor(
+            browserAdapterMock.object,
+            promiseFactory,
+            [],
```

Comment:
Can we add some test coverage for when there are known tabs ids? We can also wait and include this in the next PR, when this info actually starts persisting.

Reply:
I'll add this to the next PR, since right now there's no functional difference between passing the list to the constructor and changing the list directly after the object is initialized